cloudformation.yaml

```diff
@@ -89,6 +96,17 @@ Resources:
                     Resource:
                     - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/${DedicatedMasterAsg}
                   - !Ref AWS::NoValue
+              - !If
+                  - HasWarmDataNodes
+                  -
+                    Effect: Allow
+                    Action:
+                    - autoscaling:DetachInstances
+                    - autoscaling:AttachInstances
+                    - autoscaling:TerminateInstanceInAutoScalingGroup
+                    Resource:
+                    - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/${WarmDataNodeAsg}
+                  - !Ref AWS::NoValue
```

Comment:
Without this, we got a [Node-Rotation failure this morning](https://eu-west-1.console.aws.amazon.com/states/home?region=eu-west-1#/executions/details/arn:aws:states:eu-west-1:021353022223:execution:Elasticsearch-Node-Rotation-ophan:4f9619db-25c6-a20d-a7d4-fa77838b1798_67aeb893-81ae-5cca-5909-c0353f23de28)!